src/ctelnet.cpp

```diff
@@ -171,22 +182,76 @@ cTelnet::~cTelnet()
 
     // unicode carries information in form of single byte characters
     // and multiple byte character sequences.
     // the encoder and the decoder maintain translation state, i.e. they need to know the preceding
     // chars to make the correct decisions when translating into unicode and vice versa
 
+    // Not currently used as we do it by hand as we have to extract the data
+    // from the telnet protocol and all the out-of-band stuff.  It might be
+    // possible to use this in the future for non-UTF-8 traffic thought.
     incomingDataCodec = QTextCodec::codecForName(encoding.toLatin1().data());
     incomingDataDecoder = incomingDataCodec->makeDecoder();
 
     outgoingDataCodec = QTextCodec::codecForName(encoding.toLatin1().data());
-    outgoingDataDecoder = outgoingDataCodec->makeEncoder();
+    outgoingDataEncoder = outgoingDataCodec->makeEncoder(QTextCodec::IgnoreHeader);
+    // Do NOT create BOMs!
 }
 
+// Need to set the encoding at the start but it does not need to be written out
+// then. Return values are for Lua subsystem...!
+// newEncoding must be EITHER: one of the FIXED non-translatable values in
+// cTelnet::csmAcceptableEncodings
+// OR "ASCII"
+// OR an empty string (which means the same as the ASCII).
+QPair<bool, QString> cTelnet::setEncoding(const QString & newEncoding, const bool isToStore)
+{
+    bool isChanged = false;
+    QString reportedEncoding = newEncoding;
+    if (newEncoding.isEmpty() || newEncoding == QLatin1String("ASCII")) {
+        reportedEncoding = QLatin1String("ASCII");
+        if (! mEncoding.isEmpty()) {
+            // This will disable trancoding on:
+            // input in TBuffer::translateToPlainText(...)
+            // output in cTelnet::sendData(...)
+            mEncoding.clear();
+            if (isToStore) {
+                mpHost->writeProfileData(QLatin1String("encoding"), reportedEncoding);
+            }
+            isChanged = true;
+        }
+    } else if (!csmAcceptableEncodings.contains(newEncoding)) {
+        // Not in list - so reject it
+        return qMakePair(false,
+                         QLatin1String("(encoding \"")
+                         % newEncoding
+                         % QLatin1String("\" does not exist;\nuse one of the following:\n\"ASCII\", \"")
+                         % csmAcceptableEncodings.join(QLatin1String("\", \""))
+                         % QLatin1String("\")"));
+    } else if (mEncoding != newEncoding) {
+        encodingChanged(newEncoding);
+        isChanged = true;
+        if (isToStore) {
+            mpHost->writeProfileData(QLatin1String("encoding"), mEncoding);
+        }
+    }
+
+    if (isChanged) {
+        // Report on main console the change in ALL cases, as this can change
+        // things for the user (e.g. theoretically what characters are
+        // acceptable to the MUD Server).
+        QString message = tr("[ ALERT ] - Encoding/decoding of MUD Server traffic set to: \"%1\"...")
```

Comment:
Ellipsis is used to indicate there is more that more needs to happen before the action is complete, which isn't the case here.

How about this simpler text: `Game text encoding/decoding set to  \"%1\".` ?